compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -395,7 +395,11 @@ static void compileInternal(ModuleContext moduleContext, CompilerContext compile
         if (bootstrapLangLibName != null) {
             compilerPhaseRunner.performLangLibTypeCheckPhases(pkgNode);
         } else {
-            compilerPhaseRunner.performTypeCheckPhases(pkgNode);
+            try {
```

Comment:
This try-catch block needs to cover symbol enter phase too (line:392). For instance, https://github.com/ballerina-platform/ballerina-lang/issues/32044 is not covered with this.